### PR TITLE
VERY CRUDE downsampling. Only samples nodes & edges, not strings

### PR DESCRIPTION
**Pull request by NHDaly**

### CODE REVIEW

File: src/gc-heap-snapshot.cpp [hunk 1]
```diff
@@ -465,9 +472,84 @@ void _record_gc_just_edge(const char *edge_type, Node &from_node, size_t to_idx,
...
+            auto parent_node_pair = snapshot.node_parents.find(node_idx);
+            if (parent_node_pair == snapshot.node_parents.end()) {
+                break;
+            }
+            auto parent_idx = parent_node_pair->second;
+            auto &parent = snapshot.nodes[parent_idx];
+            node_id_to_node_map.insert(make_pair(node.id, node));
+            auto &new_node = node_id_to_node_map[node.id];
+            node_id_to_node_map.insert(make_pair(parent.id, parent));
+            auto &new_parent = node_id_to_node_map[parent.id];
```
Comment (Drvi) [inline src/gc-heap-snapshot.cpp:532]: `new_parent` is not used -- was that intended?

File: src/gc-heap-snapshot.cpp [hunk 2]
```diff
@@ -465,9 +472,104 @@ void _record_gc_just_edge(const char *edge_type, Node &from_node, size_t to_idx,
...
+            for (auto &edge : node.edges) {
+                if (edge.to_node == node_idx) {
+                    new_node.edges.push_back(edge);
+                }
```
Comment (Drvi) [inline src/gc-heap-snapshot.cpp:536]: I don't really know why we are doing this

File: src/gc-heap-snapshot.cpp (hunk 1 above)
Comment (NHDaly) [reply src/gc-heap-snapshot.cpp:532]: you're right i think it's not used. the important thing was line 531 👍

File: src/gc-heap-snapshot.cpp (hunk 2 above)
Comment (NHDaly) [reply src/gc-heap-snapshot.cpp:536]: I'm trying to rebuild the nodes from the root to the child, with edges pointing from each node along that path

**Pull request closed without merging**